kratos/processes/flux_based_redistance_process.h

```diff
@@ -0,0 +1,649 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//			 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Daniel Diez, Pablo Becker
+//
+
+#if !defined(KRATOS_POTENTIAL_FILLING_UTILITY_INCLUDED)
+#define KRATOS_POTENTIAL_FILLING_UTILITY_INCLUDED
+
+// System includes
+#include <string>
+#include <iostream>
+#include <algorithm>
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/kratos_flags.h"
+#include "elements/distance_calculation_flux_based_element.h"
+#include "geometries/geometry_data.h"
+#include "solving_strategies/schemes/residualbased_incrementalupdate_static_scheme.h"
+#include "solving_strategies/schemes/residualbased_incremental_aitken_static_scheme.h"
+#include "solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h"
+#include "solving_strategies/strategies/residualbased_newton_raphson_strategy.h"
+#include "solving_strategies/convergencecriterias/displacement_criteria.h"
+#include "processes/generic_find_elements_neighbours_process.h"
+#include "utilities/variable_utils.h"
+#include "spatial_containers/spatial_containers.h" 
+#include "modeler/connectivity_preserve_modeler.h"
+#include "utilities/merge_variable_lists_utility.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/**takes a model part full of SIMPLICIAL ELEMENTS (triangles and tetras) and computes a field which resembles the filltime
+	 * This field is to be used to compute the optimization problem of gate location
+	 * 
+	*/
+template <unsigned int TDim, unsigned int TNumNodes, class TSparseSpace, class TDenseSpace, class TLinearSolver>
+class FluxBasedRedistanceProcess : public Process
```

Comment:
As far as I see, there is some code that could be reused from previous redistance (e.g. the generation of the modelpart or the strategy). Maybe we should consider creating a base class for these.

Reply:
for the moment i'll derive the new redistance from the variational one

Reply:
OK. In the future we can refactor both to derive from the same base class.

Reply:
ok